api/v1alpha1/amaltheasession_children.go

```diff
@@ -641,3 +645,12 @@ func (as *AmaltheaSession) Secret() v1.Secret {
 	}
 	return secret
 }
+
+func (cr *AmaltheaSession) childLabels() map[string]string {
+	labels := map[string]string{}
+	maps.Copy(labels, cr.Spec.Template.Metadata.Labels)
+	// NOTE: stuff from selectorLabels will overwrite conflicts in labels (if there are any)
+	// This is the desired behaviour, we do not want to overwrite the selector labels.
+	maps.Copy(labels, selectorLabels(cr.Name))
```

Comment:
While that's what we want, should we have a warning printed so that admins can see something might be off if these labels are set ?

Reply:
@sgaist I would rather address this in a followup PR. I am still unsure of where to put this warning. And I think I will add it to the status conditions as a dedicated condition.